### PR TITLE
discuss

### PR DESCRIPTION
**Pull request by Smith-Cruise**
Fixes #issue

## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

File: be/src/formats/orc/apache-orc/c++/src/Compression.cc [hunk 1]
```diff
@@ -688,6 +689,37 @@ class BlockDecompressionStream : public DecompressionStream {
     DataBuffer<char> inputDataBuffer;
 };
 
+class LibDeflateDecompressionStream : public BlockDecompressionStream {
```
[the remark this answers is not in the document]
Comment (DorianZheng) [reply be/src/formats/orc/apache-orc/c++/src/Compression.cc:692]: %s/LibDeflate/Libdeflate/g

File: be/src/formats/orc/apache-orc/c++/src/Compression.cc [hunk 2]
```diff
@@ -1111,7 +1143,9 @@ std::unique_ptr<SeekableInputStream> createDecompressor(CompressionKind kind,
         return REDUNDANT_MOVE(input);
     case CompressionKind_ZLIB:
         return std::unique_ptr<SeekableInputStream>(
-                new ZlibDecompressionStream(std::move(input), blockSize, pool, metrics));
+                new LibDeflateDecompressionStream(std::move(input), blockSize, pool, metrics));
+//        return std::unique_ptr<SeekableInputStream>(
```
Comment (DorianZheng) [inline be/src/formats/orc/apache-orc/c++/src/Compression.cc]: don't comment useless codes

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Pull request closed without merging**